code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstanceStart.java

```diff
@@ -166,9 +167,21 @@ protected void allocate(Instance instance) {
         execute("instance.allocate", instance, null);
     }
 
+    @SuppressWarnings("unchecked")
     protected void storage(Instance instance, ProcessState state) {
         List<Volume> volumes = getObjectManager().children(instance, Volume.class);
 
+        Map<String, Object> dataVolumeMounts = DataAccessor.field(instance, DATA_VOLUME_MOUNTS, Map.class);
+
+        if (dataVolumeMounts != null) {
```

Comment:
Why don't you create a util method for this,  I think this is the 3rd time I'm seen this code.